notes/value-classes.md

```diff
@@ -422,37 +444,40 @@ fun State.updatedNow(): State {
 }
 ```
 
-This gives no improvement and is still far cry from the simplicity of working with mutable classes. So, members and extensions of value classes should support a mechanism to declare a function with a mutable receiver, that is, a function that exposes `this` receiver reference as `var` and implicitly returns the updated value of `this`. The best way to achieve it is via a modifier for a function declaration:
+This gives no improvement and is still far cry from the simplicity of working with mutable classes. So, members and extensions of value classes should support a mechanism to declare a function with a copyable receiver, that is, a function that exposes `this` receiver reference as `copy var` and implicitly returns the updated value of `this`. The best way to achieve it is via a modifier for a function declaration:
 
 ```kotlin
-mutating fun State.updateNow() { // implicitly returns new State
+copy fun State.updateNow() { // implicitly returns new State
```

Comment:
Can a `copy fun` have a return type?
I suppose no.

Reply:
But see [Mutating functions returning a value](https://github.com/Kotlin/KEEP/blob/master/notes/value-classes.md#mutating-functions-returning-a-value)

Reply:
So integration with JVM should be a different than described in "Compiling copyable properties on JVM".

Reply:
Returning a value requires a different JVM impl strategy. Either returning a pair or accepting an additional "ref" parameter. The former can be implemented efficiently with Valhalla, so that is what we'll prefer to do if we end up implementing it prior to Valhalla (since it will be easier to migrate to Valhalla later)

Reply:
Maybe you could actually do both though? like you could possibly introduce it as an annotation so that for example if you decide that returning a pair is the default one can tell the compiler to compile it as a ref instead (or compile it as both even) with an annotation (or a compiler argument) because I feel like maybe refs could be easier to optimize away with a good reducer like R8 while returned Pairs are probably harder to reason about (although I'm just guessing here with no real evidence lol)